src/plugins/impressConsole/impressConsole.js

```diff
@@ -0,0 +1,747 @@
+            root.addEventListener( 'impress:stepenter', onStepEnter );
+            root.addEventListener( 'impress:substep:stepleaveaborted', onSubstep );
+            root.addEventListener( 'impress:substep:show', onSubstepShow );
+            root.addEventListener( 'impress:substep:hide', onSubstepHide );
+
+            //When the window closes, clean up after ourselves.
+            window.onunload = function() {
+                if ( consoleWindow && !consoleWindow.closed ) {
+                    consoleWindow.close();
+                }
+            };
+
+            //Open speaker console when they press 'p'
+            registerKeyEvent( [ 80 ], open, window );
+
+            //Btw, you can also launch console automatically:
+            //<div id="impress" data-console-autolaunch="true">
+            if ( root.dataset.consoleAutolaunch === 'true' ) {
+                window.open();
+            }
+        };
+
+        var init = function( cssConsole, cssIframe ) {
+            if ( ( cssConsole === undefined || cssConsole === cssFileOldDefault ) &&
+                 ( cssIframe === undefined  || cssIframe === cssFileIframeOldDefault ) ) {
+                window.console.log( 'impressConsole.init() is deprecated. ' +
+                                   'impressConsole is now initialized automatically when you ' +
+                                   'call impress().init().' );
+            }
+            _init( cssConsole, cssIframe );
+        };
+
+        document.addEventListener( 'impress:init', function() {
+            _init();
+
+            // Add 'P' to the help popup
+            triggerEvent( document, 'impress:help:add',
+                         { command: 'P', text: 'Presenter console', row: 10 } );
+        } );
+
+        // New API for impress.js plugins is based on using events
+        root.addEventListener( 'impress:console:open', function() {
+            window.open();
+        } );
+
+        /**
+         * Register a key code to an event handler
+         *
+         * :param: event.detail.keyCodes    List of key codes
+         * :param: event.detail.handler     A function registered as the event handler
+         * :param: event.detail.window      The console window to register the keycode in
+         */
+        root.addEventListener( 'impress:console:registerKeyEvent', function( event ) {
+            registerKeyEvent( event.detail.keyCodes, event.detail.handler, event.detail.window );
+        } );
+
+        // Return the object
+        allConsoles[ rootId ] = { init: init, open: open, clockTick: clockTick,
+                               registerKeyEvent: registerKeyEvent };
+        return allConsoles[ rootId ];
+
+    };
+
+    // Returns a string to be used inline as a css <style> element in the console window.
+    // Apologies for length, but hiding it here at the end to keep it away from rest of the code.
+    var cssStyleStr = function() {
```

Comment:
The original impressConsole has external CSS files. But we don't want, or even can't, have that for impress.js plugins. So I've simply inlined the default CSS here and it is added as a <style> tag above. Not the greatest thing I ever thought of, but this is a complete window with a complex GUI, there's no way to avoid the fact it needs its own CSS.

Note the use of `backtick quoted string` below, to create a multiline string. This is an ES6 feature.